identity-appsupport/src/commonMain/kotlin/com/android/identity/appsupport/ui/certificateviewer/CertificateViewer.kt

```diff
@@ -0,0 +1,406 @@
+                                MaterialTheme.colorScheme.primary
+                            } else {
+                                MaterialTheme.colorScheme.onSurfaceVariant
+                                    .copy(alpha = .2f)
+                            }
+                        Box(
+                            modifier = Modifier
+                                .padding(2.dp)
+                                .clip(CircleShape)
+                                .background(color)
+                                .size(8.dp)
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Display certificate page data.
+ *
+ * @param data certificate content displayed on the page.
+ * @param info optional info card to display.
+ * @param warning optional warning card to display.
+ */
+@Composable
+private fun CertificateView(
+    data: CertificateViewData,
+    info: String?,
+    warning: String?,
+) {
+    Column {
+        val scrollState = rememberScrollState()
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .verticalScroll(scrollState)
+                .padding(16.dp),
+            verticalArrangement = Arrangement.spacedBy(8.dp),
+        ) {
+            info?.let { InfoCard(it) }
+            warning?.let { WarningCard(it) }
+
+            Subtitle(stringResource(Res.string.sub_basic_info))
+            KeyValuePairLine(stringResource(Res.string.k_type), data.type)
+            KeyValuePairLine(stringResource(Res.string.k_serial_number), data.serialNumber)
+            KeyValuePairLine(stringResource(Res.string.k_version), data.version)
+            KeyValuePairLine(stringResource(Res.string.k_issued), data.issued)
+            KeyValuePairLine(stringResource(Res.string.k_expired), data.expired)
+
+            if (data.subject.isNotEmpty()) {
+                Subtitle(stringResource(Res.string.sub_subject))
+                data.subject.forEach { (key, value) ->
+                    KeyValuePairLine(stringResource(resFromKey(key)), value)
+                }
+            }
+
+            if (data.issuer.isNotEmpty()) {
+                Subtitle(stringResource(Res.string.sub_issuer))
+                data.subject.forEach { (key, value) ->
+                    KeyValuePairLine(stringResource(resFromKey(key)), value)
+                }
+            }
+
+            Subtitle(stringResource(Res.string.sub_public_key_info))
+            KeyValuePairLine(stringResource(Res.string.k_pk_algorithm), data.pkAlgorithm)
+            KeyValuePairLine(stringResource(Res.string.k_pk_named_curve), data.pkNamedCurve)
+            KeyValuePairLine(stringResource(Res.string.k_pk_value), data.pkValue)
+
+            if (data.extensions.isNotEmpty()) {
+                Subtitle(stringResource(Res.string.sub_extensions))
+                data.extensions.forEach { (key, value) -> KeyValuePairLine(key, value) }
+            }
```

Comment:
Extensions are going to need a bit more work, for example conveying criticality and also pretty-printing well-known extensions. Let's do that in a subsequent PR!